game/src/edit/roads.rs

```diff
@@ -1,84 +1,461 @@
+    }
+    let available_lane_types_row = Widget::row(available_lane_types_row);
+
+    let mut current_lanes_ltr = Vec::new();
+    for (id, _, lt) in road.lanes_ltr() {
+        // TODO Add direction arrow sometimes
+        current_lanes_ltr.push(
+            ctx.style()
+                .btn_plain
+                .icon(lane_type_to_icon(lt).unwrap())
+                .disabled(Some(id) == current_lane)
+                .build_widget(ctx, format!("modify {}", id)),
+        );
+    }
+    let current_lanes_ltr = Widget::row(current_lanes_ltr);
+
+    let road_settings = Widget::row(vec![
+        Text::from_all(vec![
+            Line("Total width").secondary(),
+            Line((2.0 * road.get_half_width(map)).to_string(&app.opts.units)),
+        ])
+        .into_widget(ctx),
+        Line("Speed limit").secondary().into_widget(ctx),
+        Widget::dropdown(
+            ctx,
+            "speed limit",
+            road.speed_limit,
+            speed_limit_choices(app, Some(road.speed_limit)),
+        ),
+    ]);
+
+    Panel::new(Widget::col(vec![
+        modify_lane,
+        available_lane_types_row,
+        current_lanes_ltr,
+        road_settings,
+    ]))
+    .aligned(HorizontalAlignment::Left, VerticalAlignment::Center)
+    .build(ctx)
+}
+
+fn highlight_current_selection(
+    ctx: &mut EventCtx,
+    app: &App,
+    r: RoadID,
+    l: Option<LaneID>,
+) -> (Option<LaneID>, Drawable) {
+    let mut batch = GeomBatch::new();
+    let color = Color::hex("#DF8C3D");
+    let map = &app.primary.map;
+
+    let road = map.get_r(r);
+    batch.push(
+        color,
+        road.center_pts
+            .to_thick_boundary(2.0 * road.get_half_width(map), OUTLINE_THICKNESS)
+            .unwrap(),
+    );
+
+    if let Some(l) = l {
+        batch.push(color, app.primary.draw_map.get_l(l).get_outline(map));
+    }
+    (l, ctx.upload(batch))
+}
+
+fn lane_type_to_icon(lt: LaneType) -> Option<&'static str> {
+    match lt {
+        LaneType::Driving => Some("system/assets/edit/driving.svg"),
+        LaneType::Parking => Some("system/assets/edit/parking.svg"),
+        LaneType::Sidewalk | LaneType::Shoulder => Some("system/assets/meters/pedestrian.svg"),
+        LaneType::Biking => Some("system/assets/edit/bike.svg"),
+        LaneType::Bus => Some("system/assets/edit/bus.svg"),
+        // TODO Add an icon for this
+        LaneType::SharedLeftTurn => None,
+        LaneType::Construction => Some("system/assets/edit/construction.svg"),
+        // Don't allow creating these yet
+        LaneType::LightRail => None,
+    }
+}
+
+fn width_choices(app: &App, l: LaneID) -> Vec<Choice<Distance>> {
+    // TODO Use real standard widths for different types
+    let mut choices = vec![
+        Distance::meters(1.5),
+        Distance::meters(2.0),
+        Distance::meters(2.5),
+        Distance::meters(3.0),
+    ];
+    let current_width = app.primary.map.get_l(l).width;
+    if choices.iter().any(|x| *x != current_width) {
```

Comment:
```suggestion
    if choices.iter().all(|x| *x != current_width) {
```

Otherwise current_width is always duplicated:
<img width="325" alt="Screen Shot 2021-04-21 at 5 23 57 PM" src="https://user-images.githubusercontent.com/217057/115638046-75e37f80-a2c6-11eb-9511-73523e85a271.png">

Reply:
Good catch, thank you!